docs/guide/auto_mode.rst

```diff
@@ -0,0 +1,461 @@
+    vehicle = api.get_vehicles()[0]
+
+    # Download the current vehicle commands
+    cmds = vehicle.commands
+    cmds.download()
+    cmds.wait_valid()
+	
+    # Save the vehicle commands to a list
+    missionlist=[]
+    for cmd in cmds[1:]:  #skip first item as it is home waypoint.
+        missionlist.append(cmd)
+		
+    # Modify the mission as needed. For example, here we change the 
+    # first waypoint into a MAV_CMD_NAV_TAKEOFF command. 
+    missionlist[0].command=mavutil.mavlink.MAV_CMD_NAV_TAKEOFF
+	
+    # Clear the current mission 
+    cmds.clear()
+    vehicle.flush()
+    cmds.download()
+    cmds.wait_valid()
+	
+    #Write the modified mission and flush to the vehicle
+    for cmd in missionlist:
+        cmds.add(cmd)	
+    vehicle.flush()
+
+
+The changes are not guaranteed to be complete until 
+:py:func:`flush() <droneapi.lib.Vehicle.flush>` is called on the parent ``Vehicle`` object.
+
+
+.. _auto_mode_monitoring_controlling: 
+
+Running and monitoring missions
+===============================
+
+To start a mission change the mode to AUTO:
+
+.. code:: python
+
+    # Get an instance of the API endpoint and a vehicle
+    api = local_connect()
+    vehicle = api.get_vehicles()[0]
+
+    # Set the vehicle into auto mode
+    vehicle.mode = VehicleMode("AUTO")
+    vehicle.flush()
+
+.. note:: 
+
+    If the vehicle is in the air, then changing the mode to AUTO is all that is required to start the 
+    mission. 
+	
+    **Copter 3.3 release and later:** If the vehicle is on the ground (only), you will additionally need to send the
+    `MAV_CMD_MISSION_START <http://copter.ardupilot.com/wiki/common-mavlink-mission-command-messages-mav_cmd/#mav_cmd_mission_start>`_ 
+    command.
+
+You can stop/pause the current mission by switching out of AUTO mode (e.g. into GUIDED mode). If you switch back to 
+AUTO mode the mission will either restart at the beginning or resume at the current waypoint - the behaviour depends on the value of the 
+`MIS_RESTART <http://copter.ardupilot.com/wiki/arducopter-parameters/#mission_restart_when_entering_auto_mode_mis_restart>`_ 
+parameter (available on all vehicle types).
+
+You can monitor the progress of the mission by polling the :py:func:`Vehicle.commands.next <droneapi.lib.CommandSequence.next>` attribute
+to get the current command number. You can also change the current command by setting the attribute to the desired command number.
+
+.. code:: python
+
+    vehicle.commands.next=2
+    print "Current Waypoint: %s" % vehicle.commands.next
+    vehicle.commands.next=4
+    print "Current Waypoint: %s" % vehicle.commands.next
+
+There is no need to ``flush()`` changes to ``next`` to the vehicle (and as with other attributes, if you fetch a value, it is updated
+from the vehicle).	
+
+
+.. _auto_mode_handle_mission_end: 
+
+Handling the end of a mission
+===============================
+
+At the end of the mission the vehicle will typically "loiter" (hover in place for Copter, 
```

Comment:
Done. Removed "typically". It will always enter LOITER mode if the mission "finishes" properly. It is possible for the mission to finish "improperly" if the mission is cleared, in which case it does RTL.